client/components/shipping/packages/add-package.js

```diff
@@ -5,175 +5,177 @@ import FormLabel from 'components/forms/form-label';
 import FormTextInput from 'components/forms/form-text-input';
 import FormSettingExplanation from 'components/forms/form-setting-explanation';
 import FormButton from 'components/forms/form-button';
-import FormCheckbox from 'components/forms/form-checkbox';
+// import FormCheckbox from 'components/forms/form-checkbox';
 import Dialog from 'components/dialog';
 import AddPackagePresets from './add-package-presets';
 import { translate as __ } from 'lib/mixins/i18n';
 
-class AddPackageDialog extends React.Component {
+const getDialogButtons = ( mode, savePackage, packageData ) => {
+	return [
+		/*
+		<FormLabel className="share-package-option">
+			<FormCheckbox checked={ true } readOnly={ true } />
+			<span>Save package to use in other shipping methods</span>
+		</FormLabel>,
+		*/
+		<FormButton onClick={ () => savePackage( packageData ) }>
+			{ ( 'add' === mode ) ? __( 'Add package' ) : __( 'Apply changes' ) }
+		</FormButton>,
+	];
+};
 
-	constructor() {
-		super();
-		this.updateFormTextField = this.updateFormTextField.bind( this );
+const renderOuterDimensionsToggle = ( showOuterDimensions, packageData, toggleOuterDimensions ) => {
+	if ( ! showOuterDimensions && ! packageData.outer_dimensions ) {
+		return (
+			<a
+				href="#"
+				className="form-setting-explanation"
+				onClick={ ( evt ) => {
+					evt.preventDefault();
+					toggleOuterDimensions();
+				} }>
+				Define exterior dimensions
+			</a>
+		);
 	}
+};
 
-	getDialogButtons() {
-		const {
-			mode,
-			savePackage,
-			packageData,
-		} = this.props;
-		return [
-			<FormLabel className="share-package-option">
-				<FormCheckbox checked={ true } readOnly={ true } />
-				<span>Save package to use in other shipping methods</span>
-			</FormLabel>,
-			<FormButton onClick={ () => savePackage( packageData ) }>{ ( 'add' === mode ) ? __( 'Add package' ) : __( 'Apply changes' ) }</FormButton>,
-		];
-	}
+const updateFormTextField = ( event, updatePackagesField ) => {
+	const {
+		name,
+		value,
+	} = event.target;
+	updatePackagesField( { [name]: value } );
+};
 
-	renderOuterDimensionsToggle() {
-		const {
-			showOuterDimensions,
-			packageData,
-			toggleOuterDimensions,
-		} = this.props;
+const renderOuterDimensions = ( showOuterDimensions, packageData, value, updatePackagesField, modalReadOnly ) => {
+	return ( showOuterDimensions || packageData.outer_dimensions ) ? (
+		<FormFieldset>
+			<FormLabel>Outer Dimensions (L x W x H)</FormLabel>
+			<FormTextInput
+				name="outer_dimensions"
+				placeholder="100.25 x 25.25 x 5.75"
```

Comment:
Quebec will use commas instead of decimal points.  Noting here, but OK to fix under #303